DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -92,6 +67,50 @@ void pat::PackedCandidate::unpack() const {
       p4c.release();
     }
 }
+
+void pat::PackedCandidate::packCovariance(const reco::TrackBase::CovarianceMatrix &m, bool unpackAfterwards){
+    packedCovariance_.dptdpt = packCovarianceElement(m,0,0);
+    packedCovariance_.detadeta = packCovarianceElement(m,1,1);
+    packedCovariance_.dphidphi = packCovarianceElement(m,2,2); 
+    packedCovariance_.dxydxy =packCovarianceElement(m,3,3);
+    packedCovariance_.dzdz = packCovarianceElement(m,4,4);
+    packedCovariance_.dxydz = packCovarianceElement(m,3,4);
+    packedCovariance_.dlambdadz = packCovarianceElement(m,1,4);
+    packedCovariance_.dphidxy = packCovarianceElement(m,2,3);
+   //unpack afterwards
+   if(unpackAfterwards) unpackCovariance();
+}
+
+void pat::PackedCandidate::unpackCovariance() const {
+    const CovarianceParameterization & p=covarianceParameterization();
+    if(p.isValid()) 
+    {
+      auto m = std::make_unique<reco::TrackBase::CovarianceMatrix>() ;
+      for(int i=0;i<5;i++)
+        for(int j=0;j<5;j++){
+          (*m)(i,j)=0;
+      }
+      unpackCovarianceElement(*m,packedCovariance_.dptdpt,0,0);
+      unpackCovarianceElement(*m,packedCovariance_.detadeta,1,1);
+      unpackCovarianceElement(*m,packedCovariance_.dphidphi,2,2);
+      unpackCovarianceElement(*m,packedCovariance_.dxydxy,3,3);
+      unpackCovarianceElement(*m,packedCovariance_.dzdz,4,4);
+      unpackCovarianceElement(*m,packedCovariance_.dxydz,3,4);
+      unpackCovarianceElement(*m,packedCovariance_.dlambdadz,1,4);
+      unpackCovarianceElement(*m,packedCovariance_.dphidxy,2,3);
+      reco::TrackBase::CovarianceMatrix* expected = nullptr;
+      if( m_.compare_exchange_strong(expected,m.get()) ) {
+         m.release();
+      }
```

Comment:
it looks like there is a leak here somewhere:

from MEM_LIVE in ```pat::PackedCandidate::unpackCovariance() const```
- after 1000 events 65,704,200 bytes allocated inside the scope of this method (includes inlined methods)
    - https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_9_1_0_pre2-sign892.testDeepCSV.IgProf.1000.MEM_LIVE/99
- after 1999 events 130,948,800 bytes allocated
   - https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_9_1_0_pre2-sign892.testDeepCSV.IgProf.1999.MEM_LIVE/43

the test is based on 2K events running RecoBTag/Combined/test/testDeepCSV_cfg.py with miniAOD made in wf 10224.